### PR TITLE
[rayci] microcheck pipeline rayci config

### PR DESCRIPTION
**Pull request by can-anyscale**
As title, store this in rayci to obscure its definition.

### CODE REVIEW

File: raycicmd/config.go [hunk 1]
```diff
@@ -258,6 +259,11 @@ func ciDefaultConfig(envs Envs) *config {
 		return branchPipelineConfig
 	case rayPRPipeline, rayV2PremergePipeline, rayDevPipeline:
 		return prPipelineConfig
+	case rayV2MicrocheckPipeline:
+		config := prPipelineConfig
+		config.Env["RAYCI_MICROCHECK_RUN"] = "1"
```
Comment (aslonnie) [inline raycicmd/config.go:264]: this a map that is by reference. you would need to make a deep copy of the map

**Comment by can-anyscale** [comment]
make a copy of the struct + map; added some test cases

**Comment by can-anyscale** [comment]
convert prPipelineConfig into a function

File: raycicmd/config.go [hunk 2]
```diff
@@ -257,11 +264,16 @@ func ciDefaultConfig(envs Envs) *config {
 	case rayBranchPipeline, rayV2PostmergePipeline, rayCIPipeline:
 		return branchPipelineConfig
 	case rayPRPipeline, rayV2PremergePipeline, rayDevPipeline:
-		return prPipelineConfig
+		return prPipelineConfig("ray-pr", map[string]string{})
```
Comment (aslonnie) [inline raycicmd/config.go:267]: nit: you can just use `nil` instead of `map[string]string{}`.

**Review by aslonnie — APPROVED** [review]

**Pull request merged**